protocols/bftcosi/bftcosi.go

```diff
@@ -187,189 +185,167 @@ func (bft *ProtocolBFTCoSi) Dispatch() error {
 	}
 }
 
-// startAnnouncementPrepare create its announcement for the prepare round and
-// sends it down the tree.
-func (bft *ProtocolBFTCoSi) startAnnouncementPrepare() error {
-	ann := bft.prepare.CreateAnnouncement()
-	a := &Announce{
-		TYPE:         RoundPrepare,
-		Announcement: ann,
+// Signature will generate the final signature, the output of the BFTCoSi
+// protocol.
+func (bft *ProtocolBFTCoSi) Signature() *BFTSignature {
+	return &BFTSignature{
+		Sig:        bft.commit.Signature(),
+		Msg:        bft.Msg,
+		Exceptions: bft.tempExceptions,
+	}
+}
+
+// RegisterOnDone registers a callback to call when the bftcosi protocols has
+// really finished
+func (bft *ProtocolBFTCoSi) RegisterOnDone(fn func()) {
+	bft.onDoneCallback = fn
+}
+
+// RegisterOnSignatureDone register a callback to call when the bftcosi
+// protocol reached a signature on the block
+func (bft *ProtocolBFTCoSi) RegisterOnSignatureDone(fn func(*BFTSignature)) {
```

Comment:
Difference between `RegisterOnDone` and `RegisterOnSignatureDone` ? Once is the signature is done, the protocol still has something else to perform ?